<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd

EmployeeData = {
    'Nama' : ['Aldo', 'Richard', 'Michael'],
    'Posisi' : ['Programmer', 'System Analyst', 'Junior Data Scientist'],
    'Umur' : [25, 26, 24],
    'Gaji' : [10000000, 12000000, 9000000]
}

dfObj = pd.DataFrame(EmployeeData, index=[1,2,3])
print(dfObj)

      Nama                 Posisi  Umur      Gaji
1     Aldo             Programmer    25  10000000
2  Richard         System Analyst    26  12000000
3  Michael  Junior Data Scientist    24   9000000


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd
import io

from google.colab import files
uploaded = files.upload()

data = pd.read_csv(io.BytesIO(uploaded['diamonds-1.csv']))
print(data)


Saving diamonds-1.csv to diamonds-1 (2).csv
       Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]


Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

#Apa perbedaan Series dengan list dan dictionary
Series hanya dapat menyimpan data yang tipenya sama berbeda dengan list yang bisa menyimpan beberapa tipe data. Pada series ketika di-print dapat menampilkan indexnya sedangkan pada list tidak. Pada series index dapat diubah menjadi tipe data selain integer, sedangkan pada list tipe data selalu integer

Pada series dapat dilakukan slicing sedangkan pada dictionary tidak bisa

#Apa perbedaan dataframe dan series
Series tidak dapat mempunyai kolom ganda dan satu dimensi, sedangkan kolom pada dataframe merupakan object dari series. Dataframe berbentuk tabel/tabular


Jawab: 

In [ ]:
# code here

#Buatlah suatu data series

nama_siswa = pd.Series(['Ali',"Romi",'Diana',"Soju"])
print(nama_siswa)

0      Ali
1     Romi
2    Diana
3     Soju
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
# code here
#baris pertama dari column price
print('baris pertama dari column price')
data.head()





baris pertama dari column price


Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]

In [ ]:
#baris terakhir dari column color
print('baris terakhir dari column color')
data.tail()

baris terakhir dari column color


Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]

#baris pertama dari column price
**326**

#baris terakhir dari column color
**D**

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 
Statistika deskriptif merupakan suatu metode yang berhubungan dengan pengumpulan atau penyajian data sampai memberi informasi yang berguna.
Statistik deskriptif dapat disajikan dalam bentuk diagram, grafik, pie chart atau tabel. Informasi yang di dapatkan yang berasal dari statistika deskriptif ini antara lain ukuran pemusatan data, ukuran penyebaran data, dan juga kecenderungan suatu gugus data

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
# code here
data.describe()
#Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data


Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

Jawab:
*   Berapa nilai rata-rata dari column price? **3932.799722**
*   Berapa nilai standar deviasi dari column depth ? **1.432621**
*   Berapa nilai maximum dari column carat? **5.010000**











